### PR TITLE
workflow to automatically tag commits with version

### PR DESCRIPTION
**Pull request by PabloDons**
Version will be v$(.\gradlew -q getVersion).$workflow.run_number

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,10 +1,9 @@
 name: Release
 
 on:
+  workflow_dispatch:
   push:
     branches: [ $default-branch ]
```
Comment (paulikauro) [inline .github/workflows/release.yml]: Apparently $default-branch only gets resolved in workflow templates
```suggestion
    branches: [ master ]
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -24,20 +23,28 @@ jobs:
...
+        run:  echo ::set-output name=version::v$(./gradlew -q getVersion).${{ github.run_number }}
+
+      - name: Push new tag
+        uses: anothrNick/github-tag-action@1.26.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CUSTOM_TAG: ${{ steps.version.outputs.version }}
+
+      - name: Copy jar
         id: names
```
Comment (paulikauro) [inline .github/workflows/release.yml]: Outdated/not needed?

File: build.gradle.kts [hunk 3]
```diff
@@ -67,6 +67,9 @@ tasks.build {
     dependsOn(tasks.shadowJar)
 }
 
+tasks.register("getVersion") {
+    println(version)
+}
 
 fun execute(vararg command: String): String = ByteArrayOutputStream()
```
Comment (paulikauro) [inline build.gradle.kts]: This can be removed too now

**Review by paulikauro — CHANGES_REQUESTED** [review]
Nice, just a couple small suggestions

File: .github/workflows/release.yml (hunk 2 above)
Comment (PabloDons) [reply .github/workflows/release.yml]: It's used to generate the tag and also set the name of the jar. This is needed

File: build.gradle.kts (hunk 3 above)
Comment (PabloDons) [reply build.gradle.kts]: See other comment >:(

File: .github/workflows/release.yml (hunk 2 above)
Comment (PabloDons) [reply .github/workflows/release.yml]: Well it's a no-change and not really part of the pr anyway

Comment (paulikauro) [reply .github/workflows/release.yml]: It's just a bit weird that the name is copy jar but id is names

Comment (PabloDons) [reply .github/workflows/release.yml]: True. Maybe rename it to something else? I'm on my phone lol

Comment (paulikauro) [inline .github/workflows/release.yml]:
```suggestion
```

File: build.gradle.kts (hunk 3 above)
Comment (PabloDons) [reply build.gradle.kts]: This can't be removed btw because i need to be able to get the version from the script

Comment (PabloDons) [reply build.gradle.kts]: See line 28

Comment (paulikauro) [reply build.gradle.kts]: I mean the `execute` function

**Review by paulikauro — APPROVED** [review]

**Pull request merged**